### DISCUSSION TITLE
Ability to tell composer to use different php cli version

### DISCUSSION THREAD

**Discussion by aneeskhan47** [composer/composer#10531] [Q&A / Support]
Hi everyone,

I wanted to ask is there a way I can tell the composer to use different PHP CLI versions e.g `php8.0` instead of the default one `php` the reason for this is that some of my projects are `php7.4` and recently I'm upgrading to Laravel 9 so its minimum PHP is 8.0.

currently what I'm doing to achieve this is like this: `/usr/bin/php8.0 /usr/local/bin/composer` <- this runs composer on PHP 8.0.

I know I can make an alias of this but still, it would be cool if we can tell the `composer.json` file to use `/usr/bin/php8.0`

Thanks,
Anees

**Comment by Seldaek** [comment]
Not possible and I don't see us adding a way to do this, it'd require a process restart and sounds like a bit of a mess to have a php path hardcoded in composer.json anyway as this is very machine-dependent.
